index.js

```diff
@@ -26,6 +26,8 @@ function mergeObject(target, source, options) {
 	Object.keys(source).forEach(function(key) {
 		if (!options.isMergeableObject(source[key]) || !target[key]) {
 			destination[key] = cloneUnlessOtherwiseSpecified(source[key], options)
+		} else if (Object.keys(options.customMergeFunctions).indexOf(key) > -1) {
+			destination[key] = options.customMergeFunctions[key](source[key], target[key])
```

Comment:
source/target are switched around here, `target[key]` should be the first argument